level.py

```diff
@@ -1,44 +1,79 @@
 import ast
 
-class Level():
+
+class Level(object):
+    NULL_EVENT = (float('inf'), None)
+
     def __init__(self, filename: str=''):
-        self._events = {}
+        self._current_event = 0
+        self._events = []
 
         self._level_name = ''
         self._num_floors = 0
-        self._total_time_sec = 0.
+        self._capacity = 0
+        self._level_finish_move_number = 0
 
         if filename != '':
             self.parse_from_file(filename)
 
-    # todo implement interface for getting
+    def get_num_floors(self) -> int:
+        return self._num_floors
+
+    def get_capacity(self) -> int:
+        return self._capacity
+
+    def get_next_event(self, move: int) -> {int, [int]}:  #TODO type hinting for null event?
+        if move >= self._peek_next_event()[0]:
+            event = self._pop_next_event()
+        else:
+            event = self.NULL_EVENT
+        return event
 
-    # get_pending_event(time: float) -> Dict[int, List[int]]
+    def get_level_finish_move_number(self) -> int:
+        return self._level_finish_move_number
 
-    # get_total_time() -> float
+    def get_level_name(self) -> str:
+        return self._level_name
 
-    # get_level_name() -> str
+    def _event_get(self, i): #TODO is this suss passing back NULL_EVENT?
+        if -len(self._events) <= i < len(self._events):
+            return self._events[i]
+        else:
+            return self.NULL_EVENT
+
+    def _peek_next_event(self):
+        return self._event_get(self._current_event)
+
+    def _pop_next_event(self):
+        event = self._event_get(self._current_event)
+        self._current_event += 1
+        return event
 
     def __str__(self):
-        return "Level: {} Floors: {} Time(sec): {}\n" \
+        return "Level: {} Floors: {} Move #: {}\n" \
                "--Events--:\n{}" \
-               "".format(self._level_name, self._num_floors, self._total_time_sec, self._events)
+               "".format(self._level_name,
+                         self._num_floors,
+                         self._capacity,
+                         self._level_finish_move_number,
+                         self._events)
 
     # todo more features
     def parse_from_file(self, filename: str):
```

Comment:
parsing/creation and state should be separated at some point

Reply:
todo added